### PR TITLE
Upgrade py2 to py3

### PR DESCRIPTION
**Pull request by tbhaxor**
**Fixes issue:** #3402 

**Changes:** Changed codes from python2 syntax to python3 syntax

### CODE REVIEW

**Comment by tbhaxor** [comment]
@arnavb 
I have a time pass for you :laughing:

File: code/graph_algorithms/src/hamiltonian_cycle/hamiltonian_cycle.py [hunk 1]
```diff
@@ -109,4 +109,4 @@ def printSolution(self, path):
  
 # Print the solution
 g2.hamCycle();
-
```
Comment (arnavb) [inline code/graph_algorithms/src/hamiltonian_cycle/hamiltonian_cycle.py]: Why not just get rid of the last line?

File: code/mathematical_algorithms/src/smallest_digit_in_number/smallest_digit_in_number.py [hunk 2]
```diff
@@ -7,7 +7,7 @@ def smallest_digit(n):
         if (min_digit>n%10):
             min_digit=n%10
            
-        n=int(n/10)
+        n=n//10
```
Comment (arnavb) [inline code/mathematical_algorithms/src/smallest_digit_in_number/smallest_digit_in_number.py]: Wrap spaces around operators. Also, use `//=`.

**Comment by tbhaxor** [comment]
@arnavb review my new commits :smile:

**Comment by AdiChat** [comment]
Looks perfect 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**